qiita_db/metadata_template/base_metadata_template.py

```diff
@@ -1396,9 +1416,9 @@ def update(self, md_template):
             # Clean and validate the metadata template given
             new_map = self._clean_validate_template(
                 md_template, self.study_id, current_columns=self.categories())
-            self._update(new_map)
+            samples, columns = self._update(new_map)
             self.validate(self.columns_restrictions)
```

Comment:
Feels like it'd be more readable to test for not None or similar here, instead of having to go down into validate(), but I defer.